src/QirRuntime/experimental/ITranslator.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <sstream>
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    // Currently, there is implicit expectation that the simulator knows the runtime type of the translator and can cast
+    // to it in order to access callbacks that allow the translator to accumulate necessary information. The client must
+    // create matching (simulator, translator) pair using the corresponding factory methods, but they don't get access
+    // to the inner-workings of the translator.
+    struct ITranslator

```

Comment:
My choice of `translator` was driven by the fact that the class that implements this interface accumulates information from simulator's execution and then it can generate a new represantation of this information (that is, 'translate').